pkg/controller/core/clusterqueue_controller.go

```diff
@@ -73,6 +77,25 @@ func (r *ClusterQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	ctx = ctrl.LoggerInto(ctx, log)
 	log.V(2).Info("Reconciling ClusterQueue")
 
+	if cqObj.ObjectMeta.DeletionTimestamp.IsZero() {
+		if !controllerutil.ContainsFinalizer(&cqObj, ClusterQueueFinalizerName) {
+			controllerutil.AddFinalizer(&cqObj, ClusterQueueFinalizerName)
+			if err := r.client.Update(ctx, &cqObj); err != nil {
+				return ctrl.Result{}, client.IgnoreNotFound(err)
+			}
+		}
+	} else {
+		if controllerutil.ContainsFinalizer(&cqObj, ClusterQueueFinalizerName) {
+			if cqs, err := r.cache.AssumedWorkloads(cqObj.Name); err == nil && len(cqs) == 0 {
```

Comment:
This is not correct. 

What we need to do is the following: delay the removal of the finalizer until all running workloads are finished, not wait for assumed workloads to be bound. During this time, no new workloads should be admitted, but we should continue to update CQ status (the counters etc.); and so we need to continue to have the CQ in the cache, but in Pending state (may be we need to change the name to suspended).

Reply:
If the scheduler decides to admit a workload, then we'll assume the workload to cache. So, `r.cache.AssumedWorkloads` here is the same as the running workloads(I think the running workloads you said is equal to the admitted workloads)

Reply:
Assumed != running; AssumedWorkloads is a temporary in-memory state, once we get the update event for ```Spec.Admission```, then the  [the assumed state gets cleared](https://github.com/kubernetes-sigs/kueue/blob/14983e0126e3acb32ba4e18f76cc3c0db59cba91/pkg/cache/cache.go#L381).

Can we please revert https://github.com/kubernetes-sigs/kueue/issues/267; that change isn't needed.

Reply:
I found I made a big mistake here, I'll revert the PR.